src/client/clientmap.cpp

```diff
@@ -322,6 +328,7 @@ void ClientMap::renderMap(video::IVideoDriver* driver, s32 pass)
 	*/
 
 	MeshBufListList drawbufs;
+	std::vector<std::pair<v3s16,scene::IMeshBuffer*>> transparent_buffers;
```

Comment:
Can you use MeshBufListList here as well? Without it you'd lose the grouping that that is doing, and hence increase the number of drawcalls.

Reply:
This is intentional. Grouping meshbuffers will change the order of rendering for objects (e.g. when you have water - colored glass - water - colored glass) and produce wrong results. I will add a comment in the code that it is important to render precise mesh buffers in that order.

Reply:
Made it differently. Now MeshBufListList is only used to form the draw_order vector (previously transparent_buffers), which is not used for both solid and transparent rendering.